x/dex/keeper/contract.go

```diff
@@ -1,35 +1,40 @@
 package keeper
 
 import (
-	"encoding/binary"
 	"fmt"
 
 	"github.com/cosmos/cosmos-sdk/store/prefix"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
 )
 
 const PrefixKey = "x-wasm-contract"
 
-func (k Keeper) SetContractAddress(ctx sdk.Context, contractAddr string, codeID uint64) {
+func (k Keeper) SetContract(ctx sdk.Context, contract *types.ContractInfo) {
 	store := prefix.NewStore(
 		ctx.KVStore(k.storeKey),
 		[]byte(PrefixKey),
 	)
-	bz := make([]byte, 8)
-	binary.BigEndian.PutUint64(bz, codeID)
-	ctx.Logger().Info(fmt.Sprintf("Setting contract address %s", contractAddr))
-	store.Set(contractKey(contractAddr), bz)
+	bz, err := contract.Marshal()
+	if err != nil {
+		panic(err)
```

Comment:
why don't we return err instead of panic here

Reply:
changed to returning error